integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -663,36 +714,172 @@ public void testStandaloneManagedRestartAlways() {
+   * (b) the sample script can not stop or start a non-existing cluster
+   * (c) the sample script can not stop or start a non-existing domain.
+   */
+  @Test
+  @DisplayName("Verify that the sample script can not stop or start non-existing components")
+  public void testRestartWrongComponent() {
+    String wrongServerName = "ms1";
+    String regex = ".*" + wrongServerName + ".*\\s*is not part";
+
+    // verify that the script can not stop a non-exist server
+    String result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_SERVER_SCRIPT, SERVER_LIFECYCLE, wrongServerName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't stop a server that doesn't exist");
+
+    // verify that the script can not start a non-exist server
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(START_SERVER_SCRIPT, SERVER_LIFECYCLE, wrongServerName, "", false),
+      String.format("Failed to run %s", START_SERVER_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't start a server that doesn't exist");
+
+    // verify that the script can not stop a non-exist cluster
+    String wrongClusterName = "cluster-3";
+    regex = ".*" + wrongClusterName + ".*\\s*not part of domain";
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, wrongClusterName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(result.contains("cluster cluster-3 is not part of domain"),
+        "The script shouldn't stop a cluster that doesn't exist");
+
+    // verify that the script can not start a non-exist cluster
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(START_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, wrongClusterName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(result.contains("cluster cluster-3 is not part of domain"),
+        "The script shouldn't start a cluster that doesn't exist");
+
+    // verify that the script can not stop a non-exist domain
+    String domainName = "mii-start-policy" + "-123";
+    regex = ".*" + domainName + ".*\\s*not found";
+    result = assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_DOMAIN_SCRIPT, DOMAIN, null, "", false, domainName),
+        String.format("Failed to run %s", STOP_DOMAIN_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't stop a domain that doesn't exist");
+
+    // verify that the script can not start a non-exist domain
+    result = assertDoesNotThrow(() ->
+        executeLifecycleScript(START_DOMAIN_SCRIPT, DOMAIN, null, "", false, domainName),
+        String.format("Failed to run %s", START_DOMAIN_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't start a domain that doesn't exist");
+  }
+
+  /**
```

Comment:
Describe the configuration details with usecase as follows ...
    Currently, the domain resource has a configured cluster with two managed servers and a dynamic cluster with MaxClusterSize set to 5.

Reply:
added